src/DataTypes/DataTypeEnum.cpp

```diff
@@ -146,12 +146,22 @@ void DataTypeEnum<Type>::serializeTextEscaped(const IColumn & column, size_t row
 }
 
 template <typename Type>
-void DataTypeEnum<Type>::deserializeTextEscaped(IColumn & column, ReadBuffer & istr, const FormatSettings &) const
+void DataTypeEnum<Type>::deserializeTextEscaped(IColumn & column, ReadBuffer & istr, const FormatSettings & settings) const
 {
-    /// NOTE It would be nice to do without creating a temporary object - at least extract std::string out.
-    std::string field_name;
-    readEscapedString(field_name, istr);
-    assert_cast<ColumnType &>(column).getData().push_back(getValue(StringRef(field_name)));
+    if (settings.tsv.input_format_enum_as_number)
+    {
+        FieldType x;
+        readText(x, istr);
+        static_cast<void>(getNameForValue(x));
```

Comment:
If I understand correctly, we need this to ensure that given enum-id is correct. But this way looks a bit weird for me. Maybe it would be better to add a new void function that will do this check and raise an exception if necessary? 
@alexey-milovidov

Reply:
Yes, it will be much better.

Reply:
Thanks for the comments. I'll be able to fix it in the end of next week.